runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5312,17 +5317,15 @@ class INTERPRETER_CLASS
 					if (J9ISCONTIGUOUSARRAY(_currentThread, heapBase)) {
 						/* The address is simply the base object plus the offset. */
 						pointerValues[i] = (UDATA)heapBase + heapOffset;
-						curHeapArgIdx += 1;
 					} else {
-						/* TODO: We will need to handle the case of a discontiguous array later. */
-						buildInternalNativeStackFrame(REGISTER_ARGS);
-						updateVMStruct(REGISTER_ARGS);
-						prepareForExceptionThrow(_currentThread);
-						setCurrentExceptionUTF(_currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALARGUMENTEXCEPTION, NULL);
-						VMStructHasBeenUpdated(REGISTER_ARGS);
-						rc = GOTO_THROW_CURRENT_EXCEPTION;
-						goto done;
+						/* Copy the discontiguous array to a native memory to ensure
```

Comment:
Suggest removing "a" (... to native memory ...).

Reply:
Fixed.